### ISSUE TITLE
Hotspots using 4G internet access have droped 90% on activity the last days

### ISSUE THREAD

**Issue by rootvanished** [helium/miner#1172]
Hi, I noticed on my city that all hotspots around that I know that are using 4g Internet access have dropped 90% of the activity for the last week. Regardeless of the internet provider or the brand of the hotspot if we check activity we see that it remains many hours without any activity while Discovery mode demonstrates they can connect to more than 100 other devices around.

Please check this example: 11UcjkQyzmRQxLyZv6PFnzqY53MHvp9XqE4fXe1jcAVWzc82mD8

This hotspot was mining like 3HNT a day, it has a perfect placement and suddenly after October 24 it started dropping activity until now that it is mining around 0,1 HNT.

**Comment by rootvanished** [comment]
![witnesses chart](https://user-images.githubusercontent.com/85299777/140649677-29428cfe-937b-4da4-bd36-10a23628a28b.png)
Please check this chart reflecting the drop seen the past days.

**Comment by bitlite** [comment]
First off, check your neighbour.  1127xWx855v7Jv6UCQQDY9YcU4VNUndgBdx6a2XdthBzrcD9LXKp
He has witnesses compared to your hotspot which does not.  Therefore, you have a problem with your configuration which will no longer earn as much if you can not witness anything.

Check your LTE/4G Router and ensure port 44158 is open and confirm it via app.heliumstatus.io.

![image](https://user-images.githubusercontent.com/4667628/140768712-b89c1bcd-628f-4b44-9bb1-af0b432738e1.png)

![image](https://user-images.githubusercontent.com/4667628/140768837-59bc986f-9367-43f1-8ccc-7d0731c95f58.png)

![image](https://user-images.githubusercontent.com/4667628/140769064-a10eab4b-7fa0-4c08-ac5c-1c191bad4bfa.png)

As you can see listened link is working and the API reported link is failing.  So you may have a problem here.  Consider finding out from your provider why it is reporting a different IP address for your Network connection.  Other option, run it via VPN and hopefully it will smooth things out.

**Comment by rootvanished** [comment]
Thank you for your analysis. What I still cannot understand is why does the device start witenissing right after I switch from LTE/4G to landline fibre connectivity?

**Comment by bitlite** [comment]
Well if you are using a dual Internet connection, the Helium Miner does not constantly query the external IP address and therefore uses the one that it currently found.  This may need to be fixed by the Helium developers by adding an option in the miner that can be toggled by the miner's owner.  The option would be single or multi network option.  If multi Network option is selected, it would then query the public IP every 15 minutes to ensure it has not changed.

Request this feature request to be added so that Helium can become more robust in this area.

**Comment by rootvanished** [comment]
What do you mean by Dual internet connection?  I've setup the device in a host and I've put my mobile 4g router to provide internet to the hotspot. it worked well for 2 months. since the end of October I started see the activity decreasing so I've requested the host to temporarily use their landline fibre connectivity to understand where the problem  could reside.

With 4g I was 2 days with almost 0 witenesses, nevertheles the hotspot was creating challenges normally (I also could see from sensecap dashbaord that the internet connectivity was stable), when I switched to fibre I started getting witnesses the next hour onwards.. The device didnt even get time to gossip the new addresses and it just started showing successfull witenesses. 

I really didnt get what you meant with dual internet connection, but this is not the case.

**Comment by bitlite** [comment]
> Thank you for your analysis. What I still cannot understand is why does the device start witenissing right after I switch from LTE/4G to landline fibre connectivity?

right after I switch from LTE/4G to landline fibre connectivity? = Dual Internet connection. 
Landline Fibre connection + LTE/4G connection are 2 separate Internet connections.  If you are constantly switching between the 2, the reported API public IP will be different and will cause others to not connect correctly.  Ideally, if your LTE/4G connection is unlimited, use that instead.  Otherwise, use the fibre connection only.  No real need for dual Internet services for this.

**Comment by UrsoXone** [comment]
I am having similar issue as this HS in question. Now I can see HS has the helium API IP address matching with the public address of HS since 2 days ago, but the problem still persists..high drop in rewards. 

Any idea how can we debug this situation?

**Comment by rootvanished** [comment]
Hello, I found a workaround to get my miners to start witnessing. I've subscribed a vpn from ovpn.com with public/static ip.
I've placed this VPN in front of any internet connectivity that was not performing witenesses and suddently the hotspot starts witenessing. I also have realized that If i restart my internet router(being it 4g or landline), if the Public IP address changes then the miner (mosty of times ) starts witenessing. the difference is from having 1/2 witenesses a day to start getting 10+ perday. 

I have no idea what is the root cause and this is not restricted to 4g, also with landline I noticed this problem. What I can tell is that mantaining exacly the same setup and forcing the public IP address to change, then after 1-3 hours the hotspot comes back to normal.
